framework/framework.go

```diff
@@ -354,12 +396,13 @@ func (f *Framework) ProcessEvents(ctx context.Context, deleteChan chan watch.Eve
 	}
 
 	notifier := func(err error, d time.Duration) {
-		f.logger.Log("error", fmt.Sprintf("%#v", err))
+		f.logger.Log("warning", fmt.Sprintf("retrying operator event processing due to error: %#v", microerror.Mask(err)))
 	}
 
-	err := backoff.RetryNotify(operation, f.backOff, notifier)
+	err := backoff.RetryNotify(operation, f.backOffFactory(), notifier)
 	if err != nil {
-		f.logger.Log("error", fmt.Sprintf("%#v", err))
+		f.logger.Log("error", fmt.Sprintf("stop operator event processing retries due to too many errors: %#v", microerror.Mask(err)))
+		os.Exit(1)
```

Comment:
This is the processing of the informer events. We now use a dedicated backoff from the backoff factory and fail if this saturates the backoff. The odd thing is that the failures can happen over the course of a year or so and based on the current configuration the operator would fail as soon as the processing throws the seventh error. This is not really nice and we would need to calculate a proper retry budget where we also consider successes after errors. We should then only fail in case some error rate is reached within some timeframe. Not sure how to do this. I think we can go with this now and improve along the road in case there are real problems with this implementation.